### PR TITLE
Backports Master to 5x (5.6.10)

### PR DESCRIPTION
**Pull request by varunsharma27**
For anyone who might require it.
- Friends @sirensolutions.

### CODE REVIEW

**Comment by fbaligand** [comment]
What an amazing work !
Thanks a lot for your great contribution !
I am on holidays this week, but be sure I will review this contribution next week and merge it soon !

**Comment by varunsharma27** [comment]
No, I haven't.

**Comment by fbaligand** [comment]
Well, there are lots of breaking changes between kibana 5 and kibana 6. 
That’s the reason why there is a enhanced table 5.x branch, and why I didn’t maintained this branch until now. 
If you backport all master commits as they are, it is sure it won’t work on kibana 5. 

I suggest you to not backport commits named « add kibana 6.x support », and test it on a running kibana 5 instance.

**Comment by varunsharma27** [comment]
I've already tested all features in a running instance. Breaking changes have been dealt with.

**Comment by fbaligand** [comment]
Ok, so finally, you made functionnal tests on a running kibana 5 instance, right ?
I don’t speak about automated/coded tests. 

For info, which version did you test ?

**Comment by varunsharma27** [comment]
I manually tested it on Kibana 5.6.10

**Comment by fbaligand** [comment]
That’s great !
Said like that, I will probably merge it soon !
I will test it next week on a running kibana instance, review quickly the code, and if all is ok, merge it !

**Comment by fbaligand** [comment]
Given your first comment, you work at siren company ?

**Comment by varunsharma27** [comment]
Yes.

**Comment by fbaligand** [comment]
Great, congrats for your awesome sentinl kibana app !
Having contributed to sentinl, happy to see you contribute also to enhanced table plugin !

File: public/enhanced-table-vis.js [hunk 1]
```diff
@@ -1,32 +1,46 @@
...
+  // Vis object of this type.
   return new TemplateVisType({
+    type: 'table',
     name: 'enhanced-table',
     title: 'Enhanced Table',
     image,
     description: 'Same functionality than Data Table, but with enhanced features like computed columns and filter bar.',
-    category: VisType.CATEGORY.DATA,
-    template: visTemplate,
+    category: VisType.CATEGORY.MAP,
```
Comment (fbaligand) [inline public/enhanced-table-vis.js]: Why did you change the visualization category ?
Could you restore "DATA" category ?

Comment (varunsharma27) [reply public/enhanced-table-vis.js]: Done.

**Comment by fbaligand** [comment]
Hi @varunsharma27,

- thanks for the quick fix on visualization icon !
- I did some "manual" tests on Kibana 5.6, and globally, it works pretty well ! Thanks for the awesome work !
- Nevertheless, I detected few bugs : 
-- when I change some option (Total Function, Computed Column, Hidden columns, ...), I have to "refresh full page" (F5) so that change is applied. Press "Apply changes" is not enough.
-- when I add a computed column with "Format" = "String", I have a stack trace in browser console : `Error: this.type is undefined`

Could you look at these 2 bugs ?

**Comment by varunsharma27** [comment]
Sure!

**Comment by fbaligand** [comment]
Great, thanks !

**Comment by fbaligand** [comment]
Hi @varunsharma27,

Do you still plan to fix these 2 bugs ?
Tell me honestly, I would fully understand that you have now other priorities.
And if so, I would merge the PR and try to fix these 2 bugs for myself.

**Comment by varunsharma27** [comment]
I had some other things to sort out. I can push the fixes by the end of this week.

**Comment by fbaligand** [comment]
Hi @varunsharma27

Have you started to fix the 2 bugs ? Do you need help ? Or finally, do you have no time to continue on this PR ?

**Comment by varunsharma27** [comment]
Started working on it, the weird part being :- 
"when I change some option (Total Function, Computed Column, Hidden columns, ...), I have to "refresh full page" (F5) so that change is applied. Press "Apply changes" is not enough."
Does work on our Fork but not on standard kibana.

"when I add a computed column with "Format" = "String", I have a stack trace in browser console : Error: this.type is undefined"
Can you please give an example with "Formula (expr-eval syntax)", etc.? I'm not getting the particular error for surely col0 + col0 on String is giving out 'NaN' inside table.

**Comment by fbaligand** [comment]
Hi @varunsharma27 

As you work on sentinl plugin, you know that on each Kibana version, there are *some* breaking changes :)
So the master branch works only for Kibana 6.4.
So not every commit should be backported on 5.x branch.

To know what to keep, and what to not keep, here are some tips : 
- look original table_vis on 5.6 branch  : 
https://github.com/elastic/kibana/blob/5.6/src/core_plugins/table_vis/public/table_vis_controller.js#L27
- on enhanced table plugin commits, avoid to backport commit named "add kibana 6.XX support", which are breaking commits to support a particular new Kibana version. 
- Other commits are nice to merge. Personally, I don't rebase or merge all master commits on other branch : I cherry-pick only "feature" commits on 6.0, 6.2 and 6.3 branches.
- And,  precisely concerning "watchMulti", 'renderComplete' does not exist on Kibana 5. instead you should watch 'vis.params'.

**Comment by fbaligand** [comment]
Hi @varunsharma27,

Have you progressed on this backport ?
Tell me if you need help or if you have not anymore time to finish the backport.

**Comment by fbaligand** [comment]
Thanks for your backport work @varunsharma27 !
I just finalized the work, fixing the 2 issues I mention before.

**Comment by fbaligand** [comment]
Hi @varunsharma27 and @colmose ,

Happy to announce you I just released Kibana Enhanced Table v0.8.1 for all versions from 5.5.0 to 5.6.14!
Hope you will enjoy it!

For information, version 0.9.0 adds nothing else but Kibana 6.5 support.

**Comment by fbaligand** [comment]
You can find release packages here : 
https://github.com/fbaligand/kibana-enhanced-table/releases/tag/v0.8.1
